### PR TITLE
[perf] Only update ParameterCard when it changes

### PR DESCRIPTION
**Pull request by qnkhuat**
3b of https://github.com/metabase/metabase/issues/45780

### CODE REVIEW

File: test/metabase/models/dashboard_test.clj [hunk 1]
```diff
@@ -838,6 +839,18 @@
...
+(deftest do-not-update-parameter-card-if-it-doesn't-change-test
+  (testing "Do not update ParameterCard if updating a Dashboard doesn't change the parameters"
+    (mt/with-temp [:model/Card      {source-card-id :id} {}
+                   :model/Dashboard {dashboard-id :id} {:parameters [{:name       "Category Name"
+                                                                      :slug       "category_name"
+                                                                      :id         "_CATEGORY_NAME_"
+                                                                      :type       "category"
+                                                                      :values_source_type    "card"
+                                                                      :values_source_config {:card_id source-card-id}}]}]
+        (with-redefs [parameter-card/upsert-or-delete-from-parameters! (fn [& _] (throw (ex-info "Should not be called" {})))]
```
Comment (crisptrutski) [inline test/metabase/models/dashboard_test.clj:851]: Rather use `mt/with-dynamic-redefs`, which is thread-safe.

Comment (qnkhuat) [reply test/metabase/models/dashboard_test.clj]: ah nice! 2ab819ea80

File: test/metabase/models/card_test.clj [hunk 2]
```diff
@@ -567,6 +568,18 @@
...
+(deftest do-not-update-parameter-card-if-it-doesn't-change-test
+  (testing "Do not update ParameterCard if updating a Dashboard doesn't change the parameters"
+    (mt/with-temp [:model/Card  {source-card-id :id} {}
+                   :model/Card  {card-id-1 :id}      {:parameters [{:name       "Category Name"
+                                                                    :slug       "category_name"
+                                                                    :id         "_CATEGORY_NAME_"
+                                                                    :type       "category"
+                                                                    :values_source_type    "card"
+                                                                    :values_source_config {:card_id source-card-id}}]}]
+      (with-redefs [parameter-card/upsert-or-delete-from-parameters! (fn [& _] (throw (ex-info "Should not be called" {})))]
```
Comment (crisptrutski) [inline test/metabase/models/card_test.clj:580]: Could also update this redefs 😄

**Review by crisptrutski — APPROVED** [review]

File: test/metabase/models/card_test.clj (hunk 2 above)
Comment (qnkhuat) [reply test/metabase/models/card_test.clj:580]: :face-palm:

**Pull request merged**